src/governance-classes.cpp

```diff
@@ -0,0 +1,492 @@
+    mapTrigger.insert(make_pair(nHash, superblock));
+
+    DBG( cout << "CGovernanceTriggerManager::AddNewTrigger: End" << endl; );
+
+    return true;
+}
+
+/**
+*
+*   Clean And Remove
+*
+*/
+
+void CGovernanceTriggerManager::CleanAndRemove()
+{
+    DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: Start" << endl; );
+    AssertLockHeld(governance.cs);
+
+    // LOOK AT THESE OBJECTS AND COMPILE A VALID LIST OF TRIGGERS
+    for(trigger_m_it it = mapTrigger.begin(); it != mapTrigger.end(); ++it)  {
+        //int nNewStatus = -1;
+        CGovernanceObject* pObj = governance.FindGovernanceObject((*it).first);
+        if(!pObj)  {
+            continue;
+        }
+        CSuperblock_sptr& superblock = it->second;
+        if(!superblock)  {
+            continue;
+        }
+        // IF THIS ISN'T A TRIGGER, WHY ARE WE HERE?
+        if(pObj->GetObjectType() != GOVERNANCE_OBJECT_TRIGGER) {
+            superblock->SetStatus(SEEN_OBJECT_ERROR_INVALID);
+        }
+    }
+
+    // Remove triggers that are invalid or already executed
+    DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: mapTrigger.size() = " << mapTrigger.size() << endl; );
+    trigger_m_it it = mapTrigger.begin();
+    while(it != mapTrigger.end())  {
+        bool remove = false;
+        CSuperblock_sptr& superblock = it->second;
+        if(!superblock)  {
+            DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: NULL superblock marked for removal " << endl; );
+            remove = true;
+        }
+        else  {
+            DBG( cout << "CGovernanceTriggerManager::CleanAndRemove: superblock status = " << superblock->GetStatus() << endl; );
+            switch(superblock->GetStatus())  {
+            case SEEN_OBJECT_ERROR_INVALID:
+            case SEEN_OBJECT_UNKNOWN:
+                remove = true;
+                break;
+            case SEEN_OBJECT_EXECUTED:
+                {
+                    CGovernanceObject* govobj = superblock->GetGovernanceObject();
+                    if(govobj)  {
+                        govobj->fExpired = true;
+                    }
+                }
+                remove = true;
+                break;
+            case SEEN_OBJECT_IS_VALID:
+                {
+                    // Rough approximation: 30 days per month * 576 blocks per day
+                    static const int nMonthlyBlocks = 30*576;
+                    int nTriggerBlock = superblock->GetBlockStart();
+                    int nExpirationBlock = nTriggerBlock + nMonthlyBlocks;
+                    if(governance.GetCachedBlockHeight() > nExpirationBlock)  {
+                        remove = true;
+                        CGovernanceObject* govobj = superblock->GetGovernanceObject();
+                        if(govobj)  {
+                            govobj->fExpired = true;
+                        }
+                    }
+                }
```

Comment:
Good catch.  There's no actual impact because this is the last case before default but it should be there in case other cases are added later.